<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement():void.8dbac7d70dfe009efbda1e7016cb7873_node8</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
body { background-color:#fff; color:#000; font-family:Helvetica, sans-serif; font-size:10pt }
h1 { font-size:14pt }
.code { border-collapse:collapse; width:100%; font-family: "Andale Mono", monospace; font-size:10pt; line-height: 1.2em }
.color_black { color: black }
.color_blue { color: blue }
.color_green { color: green }
.color_orange { color: orange }
.color_red { color: red }
.line { border-left: 3px solid #ccc; padding-left: 1ex; white-space: pre; }
.msg { background-color:#fff8b4; color:#000; float:left; font-family:Helvetica, sans-serif; font-size: smaller; font-weight: bold; margin-bottom:10px; margin-top:10px; max-width:60em; padding:0.5em 1ex 0.5em 1ex; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; white-space: pre-wrap; word-wrap: break-word; }
.num { background-color:#eee; color:#444; font-size: smaller; padding-right:2ex; text-align:right; width:2.5em; }
.dangling { background-color:white; color: gray; }
.visited { background-color:LemonChiffon; color: darkmagenta; }
.tooltip { display: none; background-color:#FFF0F0; border: 2px solid #F00; font-weight: normal; left:10em; padding: 2px; position: absolute; top: -1em; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; z-index: 1}
.with_tooltip { position: relative; }
.with_tooltip:hover .tooltip, .visited:hover .tooltip { display: block; }
</style>
</head>
<body>
<center><h1>Cfg Node <a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">8</a></h1></center>PROC: <a href="./../org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873.html">void FailOnTimeoutTest.sendUpExceptionThrownByStatement()</a> LINE: <a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>
<br>PREDS:
<a href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node7.html"><span class='visited with_tooltip'>N_7<span class='tooltip'>node7 preds:6 succs:8 exn:3 </span></span></a><br>SUCCS:
<a href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node9.html"><span class='visited with_tooltip'>N_9<span class='tooltip'>node9 preds:8 succs:10 exn:3 </span></span></a><br>EXN:
<a href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node3.html"><span class='visited with_tooltip'>N_3<span class='tooltip'>node3 preds: succs:2 exn:2 </span></span></a><br>
<LISTING class='color_green'>n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70];
n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70];
*&amp;$irvar3:java.lang.Throwable*=n$12 [line 70];
</LISTING>
<hr width="100%">
<a name="session10" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session10">node8#session10</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [compute pre] liveness 8
<LISTING class='color_black'>JOIN

LEFT:   normal:{ exception, $irvar3 }, exn:{ }
RIGHT:  normal:{ }, exn:{ }
RESULT: normal:{ exception, $irvar3 }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session11" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session11">node8#session11</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] liveness 8
<LISTING class='color_black'>PRE STATE:
normal:{ exception, $irvar3 }, exn:{ }

INSTR=  *&amp;$irvar3:java.lang.Throwable*=n$12 [line 70]; 

STATE:
normal:{ n$12, exception }, exn:{ }

INSTR=  n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70]; 

STATE:
normal:{ n$11, exception }, exn:{ }

INSTR=  n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70]; 

STATE:
normal:{ exception, $irvar2 }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session31" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session31">node8#session31</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [compute pre] nullify
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session32" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session32">node8#session32</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] nullify
<LISTING class='color_black'>PRE STATE:
(normal:{ exception, $irvar2 }, exn:{ },
normal:{ n$8, n$9, n$10, $irvar1, this }, exn:{ })

INSTR=  n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70]; 

STATE:
(normal:{ n$11, exception, $irvar2 }, exn:{ },
normal:{ n$8, n$9, n$10, $irvar1, this }, exn:{ })

INSTR=  n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70]; 

STATE:
(normal:{ n$11, n$12, exception, $irvar2 }, exn:{ },
normal:{ n$8, n$9, n$10, $irvar1, this }, exn:{ })

INSTR=  *&amp;$irvar3:java.lang.Throwable*=n$12 [line 70]; 

STATE:
(normal:{ exception, $irvar3 }, exn:{ },
normal:{ n$11, n$12, $irvar2 }, exn:{ })

</LISTING>
<hr width="100%">
<a name="session52" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session52">node8#session52</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [compute pre] devirtualizer analysis 8
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session53" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session53">node8#session53</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] devirtualizer analysis 8
<LISTING class='color_black'>PRE STATE:
{ n$0 -&gt; java.lang.RuntimeException,
  n$2 -&gt; org.junit.internal.runners.statements.Fail,
  n$3 -&gt; java.lang.RuntimeException,
  n$6 -&gt; org.junit.internal.runners.statements.Fail,
  exception -&gt; java.lang.RuntimeException,
  $irvar0 -&gt; org.junit.internal.runners.statements.Fail }

INSTR=  n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70]; 

STATE UNCHANGED

INSTR=  n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70]; 

STATE UNCHANGED

INSTR=  *&amp;$irvar3:java.lang.Throwable*=n$12 [line 70]; 

STATE UNCHANGED

INSTR=  NULLIFY(&amp;$irvar2); [line 70]; 

STATE UNCHANGED

INSTR=  EXIT_SCOPE(n$11,n$12,$irvar2); [line 70]; 

STATE UNCHANGED

</LISTING>
<hr width="100%">
<a name="session74" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session74">node8#session74</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session75" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session75">node8#session75</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session76" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session76">node8#session76</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session77" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session77">node8#session77</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session78" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session78">node8#session78</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session10" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session10">node8#session10</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [compute pre] starvation
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session11" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session11">node8#session11</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] starvation
<LISTING class='color_black'>PRE STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { };
 attributes= { $irvar2 -&gt; Runnable(void FailOnTimeoutTest$3.run()) };
 thread= UnknownThread; scheduled_work= { }; var_state= { }}

INSTR=  n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { };
 attributes= { n$11 -&gt; Runnable(void FailOnTimeoutTest$3.run()),
               $irvar2 -&gt; Runnable(void FailOnTimeoutTest$3.run()) };
 thread= UnknownThread; scheduled_work= { }; var_state= { n$11 -&gt; $irvar2 }}

INSTR=  n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { };
 attributes= { n$11 -&gt; Runnable(void FailOnTimeoutTest$3.run()),
               $irvar2 -&gt; Runnable(void FailOnTimeoutTest$3.run()) };
 thread= UnknownThread; scheduled_work= { }; var_state= { n$11 -&gt; $irvar2 }}

INSTR=  *&amp;$irvar3:java.lang.Throwable*=n$12 [line 70]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { };
 attributes= { n$11 -&gt; Runnable(void FailOnTimeoutTest$3.run()),
               $irvar2 -&gt; Runnable(void FailOnTimeoutTest$3.run()) };
 thread= UnknownThread; scheduled_work= { };
 var_state= { n$11 -&gt; $irvar2, $irvar3 -&gt; n$12 }}

INSTR=  NULLIFY(&amp;$irvar2); [line 70]; 

STATE UNCHANGED

INSTR=  EXIT_SCOPE(n$11,n$12,$irvar2); [line 70]; 

STATE:
{guard_map= { }; lock_state= {map= { }; acquisitions= { }};
 critical_pairs= { }; attributes= { }; thread= UnknownThread;
 scheduled_work= { }; var_state= { }}

</LISTING>
<hr width="100%">
<a name="session29" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session29">node8#session29</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [compute pre] racerd
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session30" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session30">node8#session30</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) [exec] racerd
<LISTING class='color_black'>PRE STATE:
Threads: NoThread, Locks: 0 
Accesses { {elem= Access: Read of concurrent.TimeUnit.MILLISECONDS Thread: NoThread Lock: false Pre: Unowned;
            loc= line 232;
            trace= { FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Statement) at line 72 }},
           {elem= Access: Read of concurrent.TimeUnit.SECONDS Thread: NoThread Lock: false Pre: Unowned;
            loc= line 61;
            trace= { FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Statement) at line 72,
                     FailOnTimeout$Builder FailOnTimeout.builder() at line 231,
                     FailOnTimeout$Builder.&lt;init&gt;() at line 31 }},
           {elem= Access: Read of this-&gt;lookingForStuckThread Thread: NoThread Lock: false Pre: OwnedIf
            { 0 }; loc= line 233;
            trace= { FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Statement) at line 72 }} } 
Ownership: { n$0 -&gt; Owned,
             n$1 -&gt; Owned,
             n$2 -&gt; Owned,
             n$4 -&gt; Unowned,
             n$7 -&gt; Owned,
             n$10 -&gt; Owned,
             exception -&gt; Owned,
             this -&gt; OwnedIf{ 0 } } 
Attributes: { } 



INSTR=  n$12 := Throwable Assert.assertThrows(Class,ThrowingRunnable)({ java.lang.Exception,
                                                                    n$10 }) [line 70]; 

STATE:
Threads: NoThread, Locks: 0 
Accesses { {elem= Access: Read of concurrent.TimeUnit.MILLISECONDS Thread: NoThread Lock: false Pre: Unowned;
            loc= line 232;
            trace= { FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Statement) at line 72 }},
           {elem= Access: Read of concurrent.TimeUnit.SECONDS Thread: NoThread Lock: false Pre: Unowned;
            loc= line 61;
            trace= { FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Statement) at line 72,
                     FailOnTimeout$Builder FailOnTimeout.builder() at line 231,
                     FailOnTimeout$Builder.&lt;init&gt;() at line 31 }},
           {elem= Access: Read of this-&gt;lookingForStuckThread Thread: NoThread Lock: false Pre: OwnedIf
            { 0 }; loc= line 233;
            trace= { FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Statement) at line 72 }} } 
Ownership: { n$0 -&gt; Owned,
             n$1 -&gt; Owned,
             n$2 -&gt; Owned,
             n$4 -&gt; Unowned,
             n$7 -&gt; Owned,
             n$10 -&gt; Owned,
             n$12 -&gt; Owned,
             exception -&gt; Owned,
             this -&gt; OwnedIf{ 0 } } 
Attributes: { } 



INSTR=  NULLIFY(&amp;$irvar2); [line 70]; 

STATE UNCHANGED


</LISTING>
<hr width="100%">
<a name="session44" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session44">node8#session44</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) biabduction
<LISTING class='color_black'>**** [FP:Pending] void FailOnTimeoutTest.sendUpExceptionThrownByStatement() Node: 8, Procedure: void FailOnTimeoutTest.sendUpExceptionThrownByStatement(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_red'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_red'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
  <span class='color_red'>exception = _t$4<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_red'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>update:72</span>:void</span>; <span class='color_red'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_red'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
  [footprint
     <span class='color_red'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_red'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
  .... Instructions: ....
  <span class='color_green'>n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70];
  n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70];
  *&amp;$irvar3:java.lang.Throwable*=n$12 [line 70];
  NULLIFY(&amp;$irvar2); [line 70];
  EXIT_SCOPE(n$11,n$12,$irvar2); [line 70];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70]</span>
    in vpath_find exp:n$104
    vpath_find: cannot find n$104
    in vpath_find exp:n$100
    vpath_find do_sexp: no match on Eexp &amp;$irvar2
    vpath_find: cannot find n$100
    in vpath_find exp:n$99
    vpath_find: found exception :java.lang.Exception*
    .... Rearrangement Start ....
    Exp: &amp;$irvar2
    Prop:
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$104); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
    exception = n$99<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$101<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$102<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$100<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$103<span class='color_orange'>initial</span>:java.lang.Exception* ;
    $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException
    [footprint
       $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;$irvar2
    prop before:
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$104); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
    exception = n$99<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$101<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$102<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$100<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$103<span class='color_orange'>initial</span>:java.lang.Exception* ;
    $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException
    [footprint
       $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$104)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = n$99<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = n$101<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = n$102<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$100<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = n$103<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$3<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>_t$1<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = <span class='color_red'>n$11<span class='color_orange'>update:72</span></span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = _t$0<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70]</span>
    in vpath_find exp:n$109
    vpath_find: cannot find n$109
    in vpath_find exp:n$105
    vpath_find: found exception :java.lang.Exception*
    in vpath_find exp:n$11
    vpath_find do_sexp: no match on Eexp &amp;$irvar2
    vpath_find: cannot find n$11
    Found 2 specs for function org.junit.Assert.assertThrows(java.lang.Class,org.junit.internal.runners.statements.FailOnTimeoutTest$3):java.lang.Throwable
    START EXECUTING SPECS FOR org.junit.Assert.assertThrows(java.lang.Class,org.junit.internal.runners.statements.FailOnTimeoutTest$3):java.lang.Throwable from state
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
    [footprint
       $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    EXECUTING SPEC 1/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      [footprint
         $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>expectedThrowable|callee = _t$137<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$139<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      expectedThrowable|callee|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(String,Class,ThrowingRunnable):981 &gt; <span class='color_orange'>{vpath: }</span>(_t$140)</span> ;
      <span class='color_red'>return|callee = EXN _t$140<span class='color_orange'>update:981</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$140|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; expectedThrowable|callee|->{name:_t$138<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(n$105); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(n$11); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
    [footprint
       $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    |-
    <span class='color_blue'>expectedThrowable|callee = _t$137<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$139<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
    runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); expectedThrowable|callee|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
    
      do_imply java.lang.Exception _t$137
        Current Implication
        SUB:
          
        <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
        exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
        exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
        [footprint
           $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
        [SUB: 
          _t$137 = java.lang.Exception]
        |-
        <span class='color_blue'>runnable|callee = _t$139<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
        runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
        
          do_imply n$11 _t$139
            Current Implication
            SUB:
              
            <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
            exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
            exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
            [footprint
               $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
            [SUB: 
              _t$137 = java.lang.Exception; _t$139 = n$11]
            |-
            <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
            
              do_imply "java.lang.Exception" _t$138
              Current Implication
              SUB:
                
              <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
              exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
              exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
              [footprint
                 $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
              [SUB: 
                _t$137 = java.lang.Exception; _t$138 = "java.lang.Exception"; _t$139 = n$11]
              |-
              <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
              [FRAME TYPING:
                java.lang.Exception: java.lang.Class ]
              
                  Final Implication
                  SUB:
                    
                  <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
                  exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
                  exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException
                  [footprint
                     $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
                  [SUB: 
                    _t$137 = java.lang.Exception; _t$138 = "java.lang.Exception"; _t$139 = n$11]
                  |-
                  <span class='color_blue'></span>
                  [FRAME FLD:
                    n$11|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3]
                  [FRAME TYPING:
                    n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      [footprint
         $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
      [SUB: 
        _t$137 = java.lang.Exception; _t$138 = "java.lang.Exception"; _t$139 = n$11]
      |-
      <span class='color_blue'>expectedThrowable|callee = java.lang.Exception<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = n$11<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
      [FRAME FLD:
        n$11|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3]
      [FRAME TYPING:
        n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    returning TRUE
    in vpath_find exp:_t$140
    vpath_find: cannot find _t$140
      New footprint:
      
      Frame fld:
      n$11|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      Frame typ:
      n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class 
      Missing fld:
      
      Instantiated frame:
      n$105|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; &amp;exception|->n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; &amp;$irvar3|->n$106<span class='color_orange'>initial</span>:void; &amp;$irvar0|->0<span class='color_orange'>nullify</span>:void; &amp;return|->n$107<span class='color_orange'>initial</span>:void; &amp;$irvar1|->0<span class='color_orange'>nullify</span>:void; &amp;$irvar2|->n$11<span class='color_orange'>update:72</span>:void; &amp;this|->0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;old_this|->@f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;e|->n$108<span class='color_orange'>initial</span>:java.lang.Exception*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$140)</span> ;
      <span class='color_red'>return|callee = EXN _t$140<span class='color_orange'>update:70</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$140|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub )</span>; <span class='color_red'>n$11|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = n$106<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = n$108<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>; <span class='color_red'>java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.lang.Class</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    ***** END RESULTS FUNCTION CALL *******
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    EXECUTING SPEC 2/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      [footprint
         $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>expectedThrowable|callee = _t$141<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$142<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      expectedThrowable|callee|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(String,Class,ThrowingRunnable):981 &gt; <span class='color_orange'>{vpath: }</span>(_t$143)</span> ;
      <span class='color_red'>return|callee = EXN _t$143<span class='color_orange'>update:981</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$143|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; expectedThrowable|callee|->{}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(n$105); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(n$11); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
    [footprint
       $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
    |-
    <span class='color_blue'>expectedThrowable|callee = _t$141<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$142<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
    runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); expectedThrowable|callee|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
    
      do_imply java.lang.Exception _t$141
        Current Implication
        SUB:
          
        <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
        exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
        exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
        [footprint
           $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
        [SUB: 
          _t$141 = java.lang.Exception]
        |-
        <span class='color_blue'>runnable|callee = _t$142<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
        runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
        
          do_imply n$11 _t$142
            Current Implication
            SUB:
              
            <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
            exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
            exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
            [footprint
               $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
            [SUB: 
              _t$141 = java.lang.Exception; _t$142 = n$11]
            |-
            <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
            
                Current Implication
                SUB:
                  
                <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
                exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
                exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
                [footprint
                   $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
                [SUB: 
                  _t$141 = java.lang.Exception; _t$142 = n$11]
                |-
                <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
                [FRAME FLD:
                  java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class]
                [FRAME TYPING:
                  java.lang.Exception: java.lang.Class ]
                
                    Final Implication
                    SUB:
                      
                    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
                    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
                    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException
                    [footprint
                       $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
                    [SUB: 
                      _t$141 = java.lang.Exception; _t$142 = n$11]
                    |-
                    <span class='color_blue'></span>
                    [FRAME FLD:
                      n$11|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class]
                    [FRAME TYPING:
                      n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      [footprint
         $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
      [SUB: 
        _t$141 = java.lang.Exception; _t$142 = n$11]
      |-
      <span class='color_blue'>expectedThrowable|callee = java.lang.Exception<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = n$11<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      java.lang.Exception|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
      [FRAME FLD:
        n$11|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class]
      [FRAME TYPING:
        n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    returning TRUE
    in vpath_find exp:_t$143
    vpath_find: cannot find _t$143
      New footprint:
      
      Frame fld:
      n$11|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class
      Frame typ:
      n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class 
      Missing fld:
      
      Instantiated frame:
      n$105|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; &amp;exception|->n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; &amp;$irvar3|->n$106<span class='color_orange'>initial</span>:void; &amp;$irvar0|->0<span class='color_orange'>nullify</span>:void; &amp;return|->n$107<span class='color_orange'>initial</span>:void; &amp;$irvar1|->0<span class='color_orange'>nullify</span>:void; &amp;$irvar2|->n$11<span class='color_orange'>update:72</span>:void; &amp;this|->0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;old_this|->@f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;e|->n$108<span class='color_orange'>initial</span>:java.lang.Exception*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$143)</span> ;
      <span class='color_red'>return|callee = EXN _t$143<span class='color_orange'>update:70</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$143|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>java.lang.Exception|->{}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub )</span>; <span class='color_red'>n$11|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = n$106<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = n$108<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>; <span class='color_red'>java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    ***** END RESULTS FUNCTION CALL *******
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 2:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:java.lang.Exception*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    PROP 2 of 2:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:java.lang.Exception*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    
    Skipping instr <span class='color_green'>*&amp;$irvar3:java.lang.Throwable*=n$12 [line 70]</span> due to exception
    Instruction <span class='color_green'>NULLIFY(&amp;$irvar2); [line 70]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Exception*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    in vpath_find exp:n$11
    vpath_find do_sexp: no match on Eexp &amp;$irvar2
    vpath_find: cannot find n$11
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(n$11)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:70</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>n$11|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$11,n$12,$irvar2); [line 70]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Exception*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    in vpath_find exp:n$11
    vpath_find: cannot find n$11
    .... Prop with garbage in normal part ....
      PROP:
      <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(_t$154); MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0); UND &lt; RuntimeException.<init>() &gt; :69(@f$2); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1) ;
      exception = _t$1<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = _t$2<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = EXN _t$0<span class='color_orange'>update:70</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = 0<span class='color_orange'>nullify</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = _t$3<span class='color_orange'>initial</span>:java.lang.Exception* ;
      _t$154|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
      [footprint
         $RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub ); $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )  ]</span>
      PREDICATE:
      _t$154|->{this$0:@f$0<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:@f$1<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      ATTRIBUTE: MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>
      explain_leak: case not matched in instr <span class='color_green'>EXIT_SCOPE(n$11,n$12,$irvar2); [line 70]</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:70</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69(@f$2)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(@f$1)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:java.lang.Exception*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = @f$2<span class='color_orange'>none</span>:void( sub )</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = @f$1<span class='color_orange'>none</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>  ]</span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session54" href="./../nodes/org.junit.internal.runners.statements.FailOnTimeoutTest.sendUpExceptionThrownByStatement%28%29%3Avoid.8dbac7d70dfe009efbda1e7016cb7873_node8.html#session54">node8#session54</a>(<a href="./../../FailOnTimeoutTest.java.3b8e799a66766808563adec990adc2ea.html#LINE70">70</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void FailOnTimeoutTest.sendUpExceptionThrownByStatement() Node: 8, Procedure: void FailOnTimeoutTest.sendUpExceptionThrownByStatement(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_red'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_red'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
  <span class='color_red'>exception = _t$4<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_red'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_red'>return = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>update:72</span>:void</span>; <span class='color_red'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span></span>
  .... Instructions: ....
  <span class='color_green'>n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70];
  n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70];
  *&amp;$irvar3:java.lang.Throwable*=n$12 [line 70];
  NULLIFY(&amp;$irvar2); [line 70];
  EXIT_SCOPE(n$11,n$12,$irvar2); [line 70];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$11=*&amp;$irvar2:org.junit.function.ThrowingRunnable* [line 70]</span>
    in vpath_find exp:n$104
    vpath_find: cannot find n$104
    in vpath_find exp:n$100
    vpath_find do_sexp: no match on Eexp &amp;$irvar2
    vpath_find: cannot find n$100
    in vpath_find exp:n$99
    vpath_find: found exception :java.lang.Exception*
    .... Rearrangement Start ....
    Exp: &amp;$irvar2
    Prop:
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$104); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
    exception = n$99<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$101<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$102<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$100<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$103<span class='color_orange'>initial</span>:java.lang.Exception* ;
    $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$3<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>_t$1<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = <span class='color_red'>n$11<span class='color_orange'>update:72</span></span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = _t$0<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span></span>
    
    Instruction <span class='color_green'>n$12=_fun_Throwable Assert.assertThrows(Class,ThrowingRunnable)(java.lang.Exception:java.lang.Object*,n$11:org.junit.function.ThrowingRunnable*) [line 70]</span>
    in vpath_find exp:n$109
    vpath_find: cannot find n$109
    in vpath_find exp:n$105
    vpath_find: found exception :java.lang.Exception*
    in vpath_find exp:n$11
    vpath_find do_sexp: no match on Eexp &amp;$irvar2
    vpath_find: cannot find n$11
    Found 2 specs for function org.junit.Assert.assertThrows(java.lang.Class,org.junit.internal.runners.statements.FailOnTimeoutTest$3):java.lang.Throwable
    START EXECUTING SPECS FOR org.junit.Assert.assertThrows(java.lang.Class,org.junit.internal.runners.statements.FailOnTimeoutTest$3):java.lang.Throwable from state
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    EXECUTING SPEC 1/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>expectedThrowable|callee = _t$137<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$139<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      expectedThrowable|callee|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(String,Class,ThrowingRunnable):981 &gt; <span class='color_orange'>{vpath: }</span>(_t$140)</span> ;
      <span class='color_red'>return|callee = EXN _t$140<span class='color_orange'>update:981</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$140|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; expectedThrowable|callee|->{name:_t$138<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(n$105); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(n$11); UND &lt; RuntimeException.<init>() &gt; :69(val$3); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(val$2)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    |-
    <span class='color_blue'>expectedThrowable|callee = _t$137<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$139<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
    runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); expectedThrowable|callee|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
    
      do_imply java.lang.Exception _t$137
        Current Implication
        SUB:
          
        <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
        exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
        exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
        [SUB: 
          _t$137 = java.lang.Exception]
        |-
        <span class='color_blue'>runnable|callee = _t$139<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
        runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
        
          do_imply n$11 _t$139
            Current Implication
            SUB:
              
            <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
            exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
            exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
            [SUB: 
              _t$137 = java.lang.Exception; _t$139 = n$11]
            |-
            <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{name:_t$138<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
            
              do_imply "java.lang.Exception" _t$138
              Current Implication
              SUB:
                
              <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
              exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
              exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
              [SUB: 
                _t$137 = java.lang.Exception; _t$138 = "java.lang.Exception"; _t$139 = n$11]
              |-
              <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
              [FRAME TYPING:
                java.lang.Exception: java.lang.Class ]
              
                  Final Implication
                  SUB:
                    
                  <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
                  exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
                  exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>
                  [SUB: 
                    _t$137 = java.lang.Exception; _t$138 = "java.lang.Exception"; _t$139 = n$11]
                  |-
                  <span class='color_blue'></span>
                  [FRAME FLD:
                    n$11|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3]
                  [FRAME TYPING:
                    n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
      [SUB: 
        _t$137 = java.lang.Exception; _t$138 = "java.lang.Exception"; _t$139 = n$11]
      |-
      <span class='color_blue'>expectedThrowable|callee = java.lang.Exception<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = n$11<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>rearrange:(z)15</span>}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
      [FRAME FLD:
        n$11|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3]
      [FRAME TYPING:
        n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    returning TRUE
    in vpath_find exp:_t$140
    vpath_find: cannot find _t$140
      New footprint:
      
      Frame fld:
      n$11|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      Frame typ:
      n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class 
      Missing fld:
      
      Instantiated frame:
      n$105|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; &amp;exception|->n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; &amp;$irvar3|->n$106<span class='color_orange'>initial</span>:void; &amp;$RET_java.lang.RuntimeException.<init>()|abducedRetvar|->val$3<span class='color_orange'>formal</span>:void( sub ); &amp;$irvar0|->0<span class='color_orange'>nullify</span>:void; &amp;$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar|->val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); &amp;return|->n$107<span class='color_orange'>initial</span>:void; &amp;$irvar1|->0<span class='color_orange'>nullify</span>:void; &amp;$irvar2|->n$11<span class='color_orange'>update:72</span>:void; &amp;this|->0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;old_this|->val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;e|->n$108<span class='color_orange'>initial</span>:java.lang.Exception*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$140)</span> ;
      <span class='color_red'>return|callee = EXN _t$140<span class='color_orange'>update:70</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$140|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub )</span>; <span class='color_red'>n$11|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = n$106<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = n$108<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>; <span class='color_red'>java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.lang.Class</span></span>
    ***** END RESULTS FUNCTION CALL *******
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    EXECUTING SPEC 2/2
    ACTUAL PRECONDITION =
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    SPEC =
      --------------------------- ---------- ---------------------------
      PRE:
      <span class='color_blue'><span class='color_blue'>expectedThrowable|callee = _t$141<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$142<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      expectedThrowable|callee|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      POST 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(String,Class,ThrowingRunnable):981 &gt; <span class='color_orange'>{vpath: }</span>(_t$143)</span> ;
      <span class='color_red'>return|callee = EXN _t$143<span class='color_orange'>update:981</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$143|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; expectedThrowable|callee|->{}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span>
      ----------------------------------------------------------------
    pre_check_pure_implication
    pi1:
      MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(n$105); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(n$11); UND &lt; RuntimeException.<init>() &gt; :69(val$3); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72(val$2)
    pi2:
      
    returns
    sub1:
      
    sub2:
      
    Current Implication
    SUB:
      
    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
    |-
    <span class='color_blue'>expectedThrowable|callee = _t$141<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = _t$142<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
    runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); expectedThrowable|callee|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
    
      do_imply java.lang.Exception _t$141
        Current Implication
        SUB:
          
        <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
        exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
        exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
        [SUB: 
          _t$141 = java.lang.Exception]
        |-
        <span class='color_blue'>runnable|callee = _t$142<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
        runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
        
          do_imply n$11 _t$142
            Current Implication
            SUB:
              
            <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
            exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
            exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
            [SUB: 
              _t$141 = java.lang.Exception; _t$142 = n$11]
            |-
            <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub ); java.lang.Exception|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub )</span>
            
                Current Implication
                SUB:
                  
                <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
                exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
                exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; $irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
                [SUB: 
                  _t$141 = java.lang.Exception; _t$142 = n$11]
                |-
                <span class='color_blue'>n$11|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
                [FRAME FLD:
                  java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class]
                [FRAME TYPING:
                  java.lang.Exception: java.lang.Class ]
                
                    Final Implication
                    SUB:
                      
                    <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
                    exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
                    exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>
                    [SUB: 
                      _t$141 = java.lang.Exception; _t$142 = n$11]
                    |-
                    <span class='color_blue'></span>
                    [FRAME FLD:
                      n$11|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class]
                    [FRAME TYPING:
                      n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    Result of Abduction
      SUB:
        
      <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(runnable|callee); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
      exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; expectedThrowable|callee = java.lang.Exception<span class='color_orange'>actual_precondition</span>:java.lang.Object*; $irvar3 = n$106<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = n$107<span class='color_orange'>initial</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$11<span class='color_orange'>update:72</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); runnable|callee = n$11<span class='color_orange'>actual_precondition</span>:org.junit.function.ThrowingRunnable*; e = n$108<span class='color_orange'>initial</span>:java.lang.Exception* ;
      exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; runnable|callee|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>
      [SUB: 
        _t$141 = java.lang.Exception; _t$142 = n$11]
      |-
      <span class='color_blue'>expectedThrowable|callee = java.lang.Exception<span class='color_orange'>formal</span>:java.lang.Object*( sub ); runnable|callee = n$11<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3*( sub ) ;
      java.lang.Exception|->{}<span class='color_orange'>rearrange:(z)856</span>:java.lang.Class( sub ); runnable|callee|->{}<span class='color_orange'>rearrange:(z)1001</span>:org.junit.function.ThrowingRunnable( sub )</span>
      [FRAME FLD:
        n$11|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class]
      [FRAME TYPING:
        n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class ]
    returning TRUE
    in vpath_find exp:_t$143
    vpath_find: cannot find _t$143
      New footprint:
      
      Frame fld:
      n$11|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class
      Frame typ:
      n$11: org.junit.internal.runners.statements.FailOnTimeoutTest$3 java.lang.Exception: java.lang.Class 
      Missing fld:
      
      Instantiated frame:
      n$105|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; &amp;exception|->n$105<span class='color_orange'>update:69</span>:java.lang.Exception*; &amp;$irvar3|->n$106<span class='color_orange'>initial</span>:void; &amp;$RET_java.lang.RuntimeException.<init>()|abducedRetvar|->val$3<span class='color_orange'>formal</span>:void( sub ); &amp;$irvar0|->0<span class='color_orange'>nullify</span>:void; &amp;$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar|->val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); &amp;return|->n$107<span class='color_orange'>initial</span>:void; &amp;$irvar1|->0<span class='color_orange'>nullify</span>:void; &amp;$irvar2|->n$11<span class='color_orange'>update:72</span>:void; &amp;this|->0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;old_this|->val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); &amp;e|->n$108<span class='color_orange'>initial</span>:java.lang.Exception*
      Instantiated post:
      PROP 1 of 1:
      <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$143)</span> ;
      <span class='color_red'>return|callee = EXN _t$143<span class='color_orange'>update:70</span>:java.lang.Throwable*</span> ;
      <span class='color_red'>_t$143|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>java.lang.Exception|->{}<span class='color_orange'>formal(z)</span>:java.lang.Class( sub )</span>; <span class='color_red'>n$11|->{}<span class='color_orange'>formal(z)</span>:org.junit.function.ThrowingRunnable( sub )</span></span>
      
    ***** RESULTS FUNCTION CALL *******
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(n$109)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = n$105<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = n$106<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = n$108<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span>; <span class='color_red'>java.lang.Exception|->{name:"java.lang.Exception"<span class='color_orange'>none</span>}<span class='color_orange'>none</span>:java.lang.Class</span></span>
    ***** END RESULTS FUNCTION CALL *******
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 2:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:java.lang.Exception*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span></span>
    PROP 2 of 2:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_red'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>($irvar2)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$11<span class='color_orange'>update:72</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:java.lang.Exception*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>$irvar2|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span></span>
    
    Skipping instr <span class='color_green'>*&amp;$irvar3:java.lang.Throwable*=n$12 [line 70]</span> due to exception
    Instruction <span class='color_green'>NULLIFY(&amp;$irvar2); [line 70]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Exception*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    in vpath_find exp:n$11
    vpath_find do_sexp: no match on Eexp &amp;$irvar2
    vpath_find: cannot find n$11
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(n$11)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:70</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>n$11|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$11,n$12,$irvar2); [line 70]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$1
    vpath_find: found exception :java.lang.Exception*
    in vpath_find exp:_t$0
    vpath_find: cannot find _t$0
    in vpath_find exp:n$11
    vpath_find: cannot find n$11
    .... Prop with garbage in normal part ....
      PROP:
      <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(_t$154); MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4); MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception); MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0); UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar); UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar) ;
      exception = _t$1<span class='color_orange'>update:69</span>:java.lang.Exception*; $irvar3 = _t$2<span class='color_orange'>initial</span>:void; $RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; $RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub ); return = EXN _t$0<span class='color_orange'>update:70</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = 0<span class='color_orange'>nullify</span>:void; this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub ); e = _t$3<span class='color_orange'>initial</span>:java.lang.Exception* ;
      _t$154|->{this$0:old_this<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3; exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException; _t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
      PREDICATE:
      _t$154|->{this$0:val$1<span class='color_orange'>update:(z)72</span>, val$failOnTimeout:val$2<span class='color_orange'>update:72</span>}<span class='color_orange'>formal(z)</span>:org.junit.internal.runners.statements.FailOnTimeoutTest$3
      ATTRIBUTE: MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>
      explain_leak: case not matched in instr <span class='color_green'>EXIT_SCOPE(n$11,n$12,$irvar2); [line 70]</span>
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = _t$1<span class='color_orange'>update:69</span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = EXN _t$0<span class='color_orange'>update:70</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = _t$3<span class='color_orange'>initial</span>:java.lang.Exception*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_blue'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>MEMne &lt; ThrowingRunnable FailOnTimeoutTest.run(FailOnTimeout):72 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:70 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:69 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,FailOnTimeoutTest$3):70 &gt; <span class='color_orange'>{vpath: }</span>(_t$0)</span>; <span class='color_blue'>UND &lt; RuntimeException.<init>() &gt; :69($RET_java.lang.RuntimeException.<init>()|abducedRetvar)</span>; <span class='color_blue'>UND &lt; FailOnTimeout FailOnTimeoutTest.failAfter50Ms(Fail) &gt; :72($RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:69</span></span>:java.lang.Exception*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$2<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.RuntimeException.<init>()|abducedRetvar = val$3<span class='color_orange'>formal</span>:void( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_org.junit.internal.runners.statements.FailOnTimeoutTest.failAfter50Ms(org.junit.internal.runners.statements.Fail):org.junit.internal.runners.statements.FailOnTimeout|abducedRetvar = val$2<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeout*( sub )</span>; <span class='color_blue'>return = <span class='color_red'>EXN _t$0<span class='color_orange'>update:70</span></span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>this = 0<span class='color_orange'>nullify</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.internal.runners.statements.FailOnTimeoutTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:java.lang.Exception*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:java.lang.RuntimeException</span>; <span class='color_red'>_t$0|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    
    
  </LISTING>